src/test/java/io/github/josiasbarreto_dev/desafio_naruto/controller/CharacterControllerTest.java

```diff
@@ -0,0 +1,231 @@
+                        "Rasengan",
+                        "Sage Mode",
+                        "Kurama Chakra Mode"
+                )),
+                1000,
+                NinjaType.TAIJUTSU
+        );
+
+        payloadResponseDTO = new CharacterResponseDTO(
+                1L,
+                "Naruto Uzumaki",
+                17,
+                "Konoha",
+                new ArrayList<>(Arrays.asList(
+                        "Shadow Clone Jutsu",
+                        "Rasengan",
+                        "Sage Mode",
+                        "Kurama Chakra Mode"
+                )),
+                1000,
+                NinjaType.TAIJUTSU
+        );
+
+        payloadJutsuRequestDTO = new JutsuRequestDTO(
+                "Shouton",
+                NinjaType.TAIJUTSU
+        );
+
+        payloadChakraRequestDTO = new ChakraRequestDTO(
+                500,
+                NinjaType.NINJUTSU
+        );
+
+
+        payloadJutsuResponseDTO = new CharacterResponseDTO(
+                1L,
+                "Naruto Uzumaki",
+                17,
+                "Konoha",
+                new ArrayList<>(Arrays.asList(
+                        "Shadow Clone Jutsu",
+                        "Rasengan",
+                        "Sage Mode",
+                        "Kurama Chakra Mode",
+                        "Shouton"
+                )),
+                1000,
+                NinjaType.TAIJUTSU
+        );
+
+        payloadChakraResponseDTO = new CharacterResponseDTO(
+                1L,
+                "Naruto Uzumaki",
+                17,
+                "Konoha",
+                new ArrayList<>(Arrays.asList(
+                        "Shadow Clone Jutsu",
+                        "Rasengan",
+                        "Sage Mode",
+                        "Kurama Chakra Mode"
+                )),
+                1500,
+                NinjaType.NINJUTSU
+        );
+    }
+
+    @Test
+    @DisplayName("Deve criar um Personagem e retornar o Status Code 201")
+    void shouldCreateCharacterAndReturnSuccess() {
+        when(characterService.createCharacter(payloadRequestDTO)).thenReturn(payloadResponseDTO);
+        ResponseEntity<CharacterResponseDTO> response = characterController.createCharacter(payloadRequestDTO);
+
+        assertNotNull(response);
+        assertEquals(payloadResponseDTO, response.getBody());
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        verify(characterService, times(1)).createCharacter(payloadRequestDTO);
+    }
+
+    @Test
+    @DisplayName("Deve lançar um exceção ao criar um Personagem com Ninja Type inválido")
+    void shouldThrowExceptionWhenCreatingCharacterWithInvalidNinjaType() {
+        String message = "Ninja type is invalid";
+        when(characterService.createCharacter(payloadRequestWithNinjaTypeInvalid)).thenThrow(new IllegalArgumentException(message));
+
+        Exception exception = assertThrows(RuntimeException.class, () -> {
```

Comment:
Aqui pode ser lançado o mesmo tipo de exceção que é lançado na service
```suggestion
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
```